File: Practice-Contest-1/question3.java
import java.util.*;
public class question3 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        double r = sc.nextDouble();
        double k = sc.nextDouble();
        
        long count = 0;
        
        double top = Math.floor(r);
        if(top == r) top--;
        
        double line = Math.ceil(k);
        if(line == k) line++;
        
        double y = top;
        double x = 0;
        for(;y>=0 && y >=line;y--){
            while(y*y + x*x < r*r) x++;
            count += 2*x-1;
        }
        for(;y>=line && y>=-top;y--){
           while(y*y + x*x >= r*r) x--;
           count += 2*x+1;
        }
        
        System.out.println(count);
        
    }
}

File: Practice-Contest-1/question3.cpp
#include<bits/stdc++.h>
#include<cmath>
using namespace std;

int main(){
    double r;
    double k;
    cin>>r>>k;

    if(ceil(k) == k) k++;
    else k = ceil(k);

    double y = r;
    if(floor(y) == y) y--;
    else y = floor(y);

    long long int count = 0;
    long long x = 0;
    for(;y>=0 && y>=k;y--){
        while(y*y + x*x < r*r) x++;
        count += 2*x-1;
    }
    for(;y>-r && y>=k;y--){
        while(y*y + x*x >= r*r) x--;
        count += 2*x+1;
    }
    cout<<count<<endl;
    return 0;
}